test/test_cancel.cpp

```diff
@@ -29,7 +29,9 @@ test_1(oomph::communicator& comm, unsigned int size, int thread_id = 0)
         EXPECT_EQ(comm.scheduled_sends(), 0);
         EXPECT_EQ(comm.scheduled_recvs(), 0);
 
-        comm.send_multi(msg, dsts, 42 + 42 + thread_id).wait();
+
+        auto result = comm.send_multi(msg, dsts, 42 + 42 + thread_id);
```

Comment:
```suggestion
        comm.send_multi(msg, dsts, 42 + 42 + thread_id);
```
if we remove the nodiscard attribute, there's no need for a return value in this case (using `wait_all`)